core/src/main/java/net/opentsdb/query/processor/groupby/GroupByNumericArrayIterator.java

```diff
@@ -297,7 +297,7 @@ private void accumulateInParallel(
 
     final CountDownLatch doneSignal = new CountDownLatch(jobCount);
     for (int jobIndex = 0; jobIndex < jobCount; jobIndex++) {
-      NumericArrayAggregator combiner = combiners[jobIndex % jobCount];
+      NumericArrayAggregator combiner = combiners[jobIndex % combiners.length];
```

Comment:
we need this fix in `GroupByNumericSummaryParallelIterator` as well.

Reply:
fixed